cardano_node_tests/tests/test_governance.py

```diff
@@ -0,0 +1,347 @@
+            clusterlib_utils.cli_has("governance create-poll")
+            or clusterlib_utils.cli_has("governance answer-poll")
+            or clusterlib_utils.cli_has("governance verify-poll")
+        ):
+            pytest.skip("CLI commands `governance` poll are not available")
+
+    @pytest.fixture
+    def payment_addr(
+        self,
+        cluster_manager: cluster_management.ClusterManager,
+        cluster: clusterlib.ClusterLib,
+    ) -> clusterlib.AddressRecord:
+        """Create new payment address."""
+        amount = 200_000_000
+
+        addr = clusterlib_utils.create_payment_addr_records(
+            f"chain_tx_addr_ci{cluster_manager.cluster_instance_num}",
+            cluster_obj=cluster,
+        )[0]
+
+        # fund source address
+        clusterlib_utils.fund_from_faucet(
+            addr,
+            cluster_obj=cluster,
+            faucet_data=cluster_manager.cache.addrs_data["user1"],
+            amount=amount,
+        )
+
+        return addr
+
+    @allure.link(helpers.get_vcs_link())
+    @common.PARAM_USE_BUILD_CMD
+    def test_create_poll(
+        self,
+        cluster: clusterlib.ClusterLib,
+        payment_addr: clusterlib.AddressRecord,
+        governance_poll_available: None,  # noqa: ARG002
+        use_build_cmd: bool,
+    ):
+        """Test 'governance create-poll'.
+
+        * create the poll
+        * check that the expected outfile is created
+        * publish the poll on chain
+        * check that the created tx has the expected metadata
+        """
+        # pylint: disable=unused-argument
+        temp_template = common.get_test_id(cluster)
+
+        poll_file = f"{temp_template}_poll.json"
+
+        cli_out = cluster.cli(
+            [
+                "governance",
+                "create-poll",
+                "--question",
+                "Pineapples on pizza?",
+                "--answer",
+                "Yes",
+                "--answer",
+                "No",
+                "--out-file",
+                poll_file,
+            ]
+        )
+
+        assert "Poll created successfully" in cli_out.stderr.decode("utf-8")
+
+        # Publish poll
+        poll_metadata_file = Path(f"{temp_template}_poll_metadata.json")
+        with open(poll_metadata_file, "w", encoding="utf-8") as out_json:
+            json.dump(json.loads(cli_out.stdout.rstrip().decode("utf-8")), out_json)
+
+        tx_files = clusterlib.TxFiles(
+            signing_key_files=[
+                payment_addr.skey_file,
+                *cluster.g_genesis.genesis_keys.delegate_skeys,
+            ],
+            metadata_json_files=[poll_metadata_file],
+            metadata_json_detailed_schema=True,
+        )
+
+        if use_build_cmd:
+            tx_output = cluster.g_transaction.build_tx(
+                src_address=payment_addr.address,
+                tx_name=f"{temp_template}_publish_poll_build",
+                tx_files=tx_files,
+                witness_override=2,
```

Comment:
Please change to `witness_override=len(tx_files.signing_key_files)` as there can be different number of delegation keys for different types of cluster instances. E.g. with `babbage_fast` there are 3 delegation keys and I'm getting

> E           cardano_clusterlib.exceptions.CLIError: An error occurred running a CLI command `cardano-cli transaction submit --testnet-magic 42 --tx-file test_create_poll_ci0_kce_tx.signed --cardano-mode` on path `/home/martink/Source/repos/cardano-node-tests/tmp/pytest-of-martink/pytest-5739/test_governance_py`: Command failed: transaction submit  Error: Error while submitting tx: ShelleyTxValidationError ShelleyBasedEraBabbage (ApplyTxError [UtxowFailure (UtxoFailure (AlonzoInBabbageUtxoPredFailure (FeeTooSmallUTxO (Coin 180549) (Coin 176281))))])